crates/project/src/terminals.rs

```diff
@@ -28,10 +28,22 @@ impl Project {
             "creating terminals as a guest is not supported yet"
         );
 
+        let is_terminal = !spawn_task.is_some();
         let settings = TerminalSettings::get_global(cx);
         let python_settings = settings.detect_venv.clone();
         let (completion_tx, completion_rx) = bounded(1);
         let mut env = settings.env.clone();
+
+        // if we are creating a task, activate minimal python virtual environment
+        if !is_terminal {
```

Comment:
We can't move them both because the full venv activation should commence after the terminal is created and `set_python_venv_path_for_tasks` is done before. On the other hand, I did in fact overlook the `if let Some(spawn_task) = spawn_task {` branch, so I'd move the minimal venv activation there